test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -0,0 +1,148 @@
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+
+import static javax.swing.SwingUtilities.invokeAndWait;
+import static javax.swing.SwingUtilities.isEventDispatchThread;
+
+public class PassFailJFrame extends JFrame {
+    private final CountDownLatch latch = new CountDownLatch(1);
+    private boolean failed = false;
+    private String testFailedReason;
+    private JTextArea instructionsText;
+    private final int maxRowLength;
+    private final int maxStringLength;
+    private final int timeoutMinutes;
+
+    /**
+     * Constructs a JFrame with a given title & serves as test instructional
+     * frame where the user follows the specified test instruction in order
+     * to test the test case & mark the test pass or fail. If the expected
+     * result is seen then the user click on the 'Pass' button else click
+     * on the 'Fail' button and the reason for the failure should be
+     * specified in the JDailog JTextArea.
+     *
+     * @param title           title of the Frame.
+     * @param instructions    specified instruction that user should follow.
+     * @param maxRowLength    number of visible rows of the JTextArea where the
+     *                        instruction is show.
+     * @param maxStringLength number of columns of the instructional JTextArea
+     * @param timeoutMinutes  timeout of the test where time is specified in
+     *                        minutes.
+     * @throws HeadlessException
+     * @throws InterruptedException
+     * @throws InvocationTargetException
+     */
+    public PassFailJFrame(String title, String instructions,
+                          int maxRowLength, int maxStringLength,
+                          int timeoutMinutes) throws HeadlessException,
+            InterruptedException, InvocationTargetException {
+        super(title);
+        this.maxRowLength = maxRowLength;
+        this.maxStringLength = maxStringLength;
+        this.timeoutMinutes = timeoutMinutes;
+
+        invokeAndWait(() -> {
+            setLayout(new BorderLayout());
+            instructionsText = new JTextArea(instructions, maxRowLength, maxStringLength);
+            instructionsText.setEditable(false);
+            instructionsText.setFocusable(false);
+            add(instructionsText, BorderLayout.NORTH);
+
+            JButton btnPass = new JButton("Pass");
+            btnPass.addActionListener((e) -> latch.countDown());
+
+            JButton btnFail = new JButton("Fail");
+            btnFail.addActionListener((e) -> getFailureReason());
+
+            JPanel buttonsPanel = new JPanel();
+            buttonsPanel.add(btnPass);
+            buttonsPanel.add(btnFail);
+
+            add(buttonsPanel, BorderLayout.SOUTH);
+            pack();
+            setLocation(10, 10);
```

Comment:
The instruction window should probably be placed around the centre of the screen.

The class should provide a method to position a secondary frame near the instruction frame so that both frames don't overlap.

A possibility to include test UI to the instructions could be beneficial in some cases: dealing with one window is easier than with several ones. It's just a suggestion for further improvement though.

Reply:
Fixed